nats_test.go

```diff
@@ -2718,6 +2718,49 @@ X-Test-Keys: F
 	}
 }
 
+func BenchmarkHeaderDecode(b *testing.B) {
```

Comment:
Please move this benchmark to `bench_test.go`

Reply:
`decodeHeadersMsg` is private, and is only called by other private functions so it's going to be hard to isolate this from an external package, do you have any suggestions?

Reply:
Yeah, you're right, sorry for confusion. Not sure if it makes sense to create a separate `bench_test.go` in `nats` package just for the sake of a single benchmark, so lets leave it where it is for now (maybe you could move it to the bottom of the file to separate it from other tests).

Reply:
I've moved to the bottom as suggested